src/Orleans.Core.Abstractions/IDs/UniqueKey.cs

```diff
@@ -287,44 +288,48 @@ internal uint GetUniformHashCode()
             // ReSharper restore NonReadonlyFieldInGetHashCode
         }
 
+        /// <summary>
+        /// If KeyExt not exists, returns following structure
+        /// |8 bytes|8 bytes|8 bytes|4 bytes| - total 28 bytes.
+        /// If KeyExt exists, adds additional KeyExt bytes length
+        /// </summary>
+        /// <returns></returns>
         internal byte[] ToByteArray()
         {
-            byte[] bytes, extBytes = null;
-            var tmpArray = new ulong[1];
+            var extBytes = this.KeyExt != null ? Encoding.UTF8.GetBytes(KeyExt) : null;
```

Comment:
I believe that these optimizations would benefit greatly from .NET Core 2.1/3.0 specialization (via multi-targeting).

What do you think, Igor?

Multi-targeting is something I think we will should start doing soon, even though it has some downsides

Reply:
I think in this case, we can avoid complexity of maintaining multitargeting, because as i showed benchmak results above, slow spans have almost same efficiency as native spans.